File: src-tauri/src/dto/pmid_dto.rs
use serde::Serialize;

#[derive(Debug,Serialize)]
#[serde(rename_all = "camelCase")]
pub struct PmidDto {
    /// Full PMID with no whitespace, e.g. 'PMID:12345'
    pub pmid: String,
    /// Title of corresponding article
    pub title: String,
}

impl PmidDto {
    pub fn new(pmid: &str, title: &str) -> Self {
        Self {
            pmid: pmid.to_string(),
            title: title.to_string()
        }
    }

    pub fn from_numerical_pmid(num_pmid: &str, title: &str) -> Self {
        let pmid = format!("PMID:{}", num_pmid);
        Self::new(&pmid, title)
    }
}

File: src/app/models/pmid_dto.ts
export interface PmidDto {
    pmid: string,
    title: string,
    hasError: boolean,
    retrievedPmid: boolean,
    errorMessage: string
}

export function defaultPmidDto(): PmidDto {
    return {
        pmid: '',
        title: '',
        hasError: false,
        retrievedPmid: false,
        errorMessage: '',
    };
}
